seatunnel-engine/seatunnel-engine-server/src/main/java/org/apache/seatunnel/engine/server/resourcemanager/allocation/strategy/SlotRatioStrategy.java

```diff
@@ -0,0 +1,83 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.seatunnel.engine.server.resourcemanager.allocation.strategy;
+
+import org.apache.seatunnel.engine.server.resourcemanager.worker.WorkerProfile;
+
+import org.apache.commons.lang3.tuple.ImmutableTriple;
+
+import com.hazelcast.cluster.Address;
+
+import java.util.Comparator;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+/** SlotRatioStrategy is a strategy that selects the worker with the lowest slot usage rate. */
+public class SlotRatioStrategy implements SlotAllocationStrategy {
+
+    private final Map<Address, ImmutableTriple<Double, Integer, Integer>> workerAssignedSlots;
+
+    public SlotRatioStrategy(
+            Map<Address, ImmutableTriple<Double, Integer, Integer>> workerAssignedSlots) {
```

Comment:
I can't know the ImmutableTriple values used for. How about use a object to store it?

Reply:
ok，I will currently refactoring